stvisualizer/packages/vs-syntax-tree/src/representations/dropdown-tree.tsx

```diff
@@ -0,0 +1,68 @@
+"use strict";
+/**
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+import React, { useEffect, useState } from "react";
+import DropdownNode from "../components/dropdown/dropdownNode";
+import DropdownNodeDetails from "../components/dropdown/dropdownNodeDetails";
+import { DropdownTreeProps, TreeNodeObject } from "../resources/tree-interfaces";
+import * as styles from "../styles/dropdown-tree.styles";
+
+function DropdownTree(props: DropdownTreeProps) {
+    const {treeNode, onCollapseTree, onFindNode} = props;    
+    const [detailedNode, setDetailedNode] = useState<TreeNodeObject | undefined>(undefined);
+
+    useEffect(() => {
+        setDetailedNode(treeNode);
+    }, [treeNode.nodeID]);
```

Comment:
Wouldn't adding just `[treeNode]` suffice here?

Reply:
Since it is an object, the behaviour deviates without the specific property check.

Reply:
:+1: No problem then. Usually the concept is, we should add whatever we use within the `useEffect()` callback as a dependency. My only concern is having `treeNode.nodeID` as the dependency while using `treeNode` inside. In the implementation, can the `treeNode.nodeID` change without changing the `treeNode` (i.e. without changing object reference).

Reply:
The other properties of the object can change without the ID being changed. But we wont this to trigger only if and when the ID changes, i.e., if the syntax tree updates with file changes then the nodes are allocated new IDs.